merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/engine/SimulationEngine.java

```diff
@@ -503,6 +503,23 @@ public SimulationResults computeResults(
                                  serializedTimeline);
   }
 
+  public Duration getTaskDuration(String id){
```

Comment:
If we're able to figure out whatever was [the problem here](https://github.com/NASA-AMMOS/aerie/pull/23#discussion_r783570163), this would be better to take a `TaskId`. The use of `String` as the concrete ID type is mostly an implementation detail, modulo some unfortunate boundary-crossing with the return type of `spawn()`.

(You're right -- the search under this line really is very unfortunate.)

Reply:
You will see I have anticipated the change of `spawn` and directly used one of the `initiate` method to get the `TaskId`. Also got rid of the `ControlTask`.

Reply:
And also replaced this `String` id by`TaskId`.